File: 백준/Silver/1260.DFS와 BFS/DFS와 BFS.java
import java.util.*;
import java.io.*;

public class Main {
    static ArrayList<ArrayList<Integer>> graph = new ArrayList<ArrayList<Integer>>();
    static boolean[] visited;

    public static void bfs(int start) {
        Queue<Integer> q = new LinkedList<>();
        q.offer(start);
        visited[start] = true;

        while(!q.isEmpty()){
            int x = q.poll();
            System.out.print(x + " ");
            for(int i = 0; i < graph.get(x).size();i++){
                int y = graph.get(x).get(i);
                if(!visited[y]){
                    q.offer(y);
                    visited[y] = true;
                }
            }
        }
    }

    public static void dfs(int start) {
        visited[start] = true;
        System.out.print(start + " ");
        for(int i = 0; i < graph.get(start).size();i++){
            int x = graph.get(start).get(i);
            if(!visited[x]) dfs(x);
        }
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = null;

        st = new StringTokenizer(br.readLine());
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());
        int s = Integer.parseInt(st.nextToken());


        for (int i = 0; i <= n ; i++) {
            graph.add(new ArrayList<Integer>());
        }

        for (int i = 0; i < m; i++) {
            st = new StringTokenizer(br.readLine());
            int a = Integer.parseInt(st.nextToken());
            int b = Integer.parseInt(st.nextToken());

            graph.get(a).add(b);
            graph.get(b).add(a);
        }
        for(int i = 0; i <= n;i++){
            Collections.sort(graph.get(i));
        }

        visited = new boolean[1001];
        dfs(s);
        System.out.println();

        visited = new boolean[1001];
        bfs(s);
    }
}

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py
import sys
from collections import deque


def dfs(graph, v, visited):
    visited[v] = True
    print(v, end=' ')

    for i in graph[v]:
        if not visited[i]:
            dfs(graph, i, visited)


def bfs(graph, v, visited):
    q = deque()
    q.append(v)

    while q:
        x = q.popleft()
        visited[x] = True
        print(x, end=' ')
        for i in graph[x]:
            if not visited[i]:
                visited[i] = True
                q.append(i)


n, m, v = map(int, sys.stdin.readline().split())

graph = [[] for _ in range(n + 1)]

for _ in range(m):
    a, b = map(int, sys.stdin.readline().split())
    graph[a].append(b)
    graph[b].append(a)

visited = [False] * (n + 1)

for i in range(1, n + 1):
    graph[i].sort()

dfs(graph, v, visited)
print()
visited = [False] * (n + 1)
bfs(graph,v,visited)